src/common/mfu_flist_walk.c

```diff
@@ -657,11 +668,15 @@ void mfu_flist_walk_paths(uint64_t num_paths, const char** paths,
         CIRCLE_cb_process(&walk_stat_process);
     }
     else {
-        /* walk directories using file types in readdir */
-        CIRCLE_cb_create(&walk_readdir_create);
-        CIRCLE_cb_process(&walk_readdir_process);
-        //        CIRCLE_cb_create(&walk_getdents_create);
-        //        CIRCLE_cb_process(&walk_getdents_process);
+        if (walk_opts->no_atime) {
+            /* walk directories without updating the file last access time */
+            CIRCLE_cb_create(&walk_getdents_create);
```

Comment:
I agree this if/else block works, but what's the value in having both `walk_getdents_create()` and `walk_readdir_create()`?  Their code is identical, it seems, except for whether they call `walk_getdents_process_dir()` or `walk_readdir_process_dir()`.

Reply:
I agree with you, I hadn't realized that the functions `walk_getdents_create()` and `walk_readdir_create()` were identical. The same question applies to the functions `walk_getdents_process` and `walk_readdir_process` which are also identical except for whether they call `walk_readdir_process_dir()` or `walk_getdents_process_dir()`.

I think it's best to make a generic function that will call the desired function depending on whether you have the noatime option. This way, the if/else block is no longer necessary.
`